shared/actions/unlock-folders.js

```diff
@@ -80,19 +67,17 @@ export function registerRekeyListener (): (dispatch: Dispatch) => void {
     })
 
     // we get this with sessionID == 0 if we call openDialog
-    engine.listenGeneralIncomingRpc({
-      'keybase.1.rekeyUI.refresh': (params, response) => refreshHandler(params, response, dispatch),
-    })
+    engine.setIncomingHandler('keybase.1.rekeyUI.refresh', (params, response) => refreshHandler(params, response, dispatch))
+
     // else we get this also as part of delegateRekeyUI
-    rekeyServer = createServer(
-      engine,
-      'keybase.1.rekeyUI.delegateRekeyUI',
-      null,
-      () => ({
-        'keybase.1.rekeyUI.delegateRekeyUI': (params, response) => { },
+    engine.setIncomingHandler('keybase.1.rekeyUI.delegateRekeyUI', (param: any, response: ?Object) => {
```

Comment:
instead of the whole server layer on top of engine we simply listen for this message and create a new session